src/repository/index.ts

```diff
@@ -122,10 +129,23 @@ export default class Repository extends EventEmitter implements EventEmitter {
     this.bootstrapOverride = bootstrapOverride;
     this.storageProvider = storageProvider;
     this.segments = new Map();
+    this.streaming = streaming;
+    if (this.streaming) {
+      this.eventSource = new EventSource(resolveUrl(this.url, './client/streaming'), {
+        headers: this.headers,
+      } as any);
+      this.eventSource?.addEventListener('message', (event) => {
```

Comment:
required EventSource is seen as potentially undefined. Not sure why that's what ? is handling here